src/main/java/sg/ncl/MainController.java

```diff
@@ -232,40 +232,51 @@ public String loginSubmit(
         byte[] base64CredsBytes = Base64.encodeBase64(plainCredsBytes);
         String base64Creds = new String(base64CredsBytes);
 
-        ResponseEntity responseEntity;
+        ResponseEntity response;
         String jwtTokenString;
         String id = "";
 
-        try {
-            HttpHeaders headers = new HttpHeaders();
-            headers.set("Authorization", "Basic " + base64Creds);
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Basic " + base64Creds);
 
-            HttpEntity<String> request = new HttpEntity<>(headers);
-            responseEntity = restTemplate.exchange(properties.getSioAuthUrl(), HttpMethod.POST, request, String.class);
-            jwtTokenString = responseEntity.getBody().toString();
+        HttpEntity<String> request = new HttpEntity<>(headers);
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+
+        try {
+            response = restTemplate.exchange(properties.getSioAuthUrl(), HttpMethod.POST, request, String.class);
+            jwtTokenString = response.getBody().toString();
         } catch (Exception e) {
-            // TODO: handle different types of error
-            // case1: invalid login
-            loginForm.setErrorMsg("Login failed: Invalid email/password.");
+            logger.warn("Error connecting to authentication service to validate login details");
+            loginForm.setErrorMsg(ERR_SERVER_OVERLOAD);
             return "login";
         }
 
-        if (jwtTokenString != null || !jwtTokenString.isEmpty()) {
-            JSONObject tokenObject = new JSONObject(jwtTokenString);
-            String token = tokenObject.getString("token");
-            id = tokenObject.getString("id");
-
-            logger.info("login success with token id {}", id);
-            AUTHORIZATION_HEADER = "Bearer " + token;
-
-            session.setAttribute("sessionLoggedEmail", loginForm.getLoginEmail());
-            session.setAttribute("id", id);
-
-            return "redirect:/dashboard";
-        } else {
-            // case1: invalid login
-            loginForm.setErrorMsg("Login failed: Invalid email/password.");
-            return "login";
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(jwtTokenString, MyErrorResource.class);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "error" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "error". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1854)